upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1027,6 +956,82 @@ func (c *ApplyClusterCmd) validateKubernetesVersion() error {
 	return nil
 }
 
+// addFileAssets adds the file assets within the assetBuilder
+func (c *ApplyClusterCmd) addFileAssets(assetBuilder *assets.AssetBuilder) error {
+
+	var baseURL string
+	var err error
+	if components.IsBaseURL(c.Cluster.Spec.KubernetesVersion) {
+		baseURL = c.Cluster.Spec.KubernetesVersion
+	} else {
+		baseURL = "https://storage.googleapis.com/kubernetes-release/release/v" + c.Cluster.Spec.KubernetesVersion
+	}
+
+	k8sAssetsNames := []string{
+		"/bin/linux/amd64/kubelet",
+		"/bin/linux/amd64/kubectl",
+	}
+	if needsKubernetesManifests(c.Cluster, c.InstanceGroups) {
+		k8sAssetsNames = append(k8sAssetsNames, "/kubernetes-manifests.tar.gz")
+	}
+
+	for _, a := range k8sAssetsNames {
+		k, err := url.Parse(baseURL)
+		if err != nil {
+			return err
+		}
+		k.Path = path.Join(k.Path, a)
+
+		u, hash, err := assetBuilder.RemapFileAndSHA(k)
+		if err != nil {
+			return err
+		}
+		c.Assets = append(c.Assets, hash.Hex()+"@"+u.String())
+	}
+
+	if usesCNI(c.Cluster) {
+		cniAsset, cniAssetHashString, err := findCNIAssets(c.Cluster, assetBuilder)
+		if err != nil {
+			return err
+		}
+
+		c.Assets = append(c.Assets, cniAssetHashString+"@"+cniAsset.String())
+	}
+
+	// TODO figure out if we can only do this for CoreOS
+	{
+		utilsLocation, hash, err := FileUrl("linux/amd64/utils.tar.gz", assetBuilder)
+		if err != nil {
+			return err
+		}
+		c.Assets = append(c.Assets, hash.Hex()+"@"+utilsLocation.String())
+	}
+
+	// FIXME why does using the hash break Cloudformation integration test?
+	//n, hash, err := NodeUpLocation(assetBuilder)
+	//c.NodeUpHash = hash.Hex()
+	n, _, err := NodeUpLocation(assetBuilder)
+	if err != nil {
+		return err
+	}
+	c.NodeUpSource = n.String()
+
+	// When the Target is DryRun this is not being called
```

Comment:
Suggestion: 
```
// We explicitly include this here, because otherwise it is not included as an asset when Target is DryRun
// Is there another way to call this?
``` 

assuming that's correct!